### DISCUSSION TITLE
Digitally signed PDFs

### DISCUSSION THREAD

**Discussion by coolbung** [egovernments/Digit-Core#576] [DIGIT Platform]
Do any of the DIGIT Core services allow creation of digitally signed PDF files (using PKI) ?

If not, what would be a broad solution for enabling this.

**Comment by talele08** [comment]
Currently, we do not have a service that can digitally sign PDFs using PKI. You can generate the PDF using DIGIT's PDF service, sign it externally, and then upload it to DIGIT's filestore service. While we have a sign API in the encryption service, it currently generates keys internally for each tenant, which may not align with your PKI requirements.

**Reply by suresh12** [reply]
@talele08 We do have scenario where we need to sign an existing PDF. We can't send the file to eSign service. We need to send the file hash to eSign service which will provide a signature, which needs to be embedded in the file. Need to make sure filehash is not changed when the signature is embedded. Since PDF service does not allow modification, to embed signature you need to generate a PDF which change the hash.
